src/name.cpp

```diff
@@ -1,164 +1,154 @@
 #include <map>
-#include <sstream>
+#include <string>
 
-#include "json.h"
 #include "name.h"
+#include "json.h"
 #include "string_formatter.h"
-#include "output.h"
 #include "translations.h"
 #include "rng.h"
 #include "cata_utility.h"
 
-NameGenerator::NameGenerator()
+static inline nameFlags operator&( nameFlags l, nameFlags r )
 {
-
+    return static_cast<nameFlags>( static_cast<unsigned>( l ) & static_cast<unsigned>( r ) );
 }
 
-void NameGenerator::clear_names()
+namespace Name
 {
-    names.clear();
+static std::map< nameFlags, std::vector< std::string > > names;
+
+struct str_flag {
+    char const *str;
+    nameFlags f;
+};
+
+static const str_flag usage_flags[] = {
+    { "given",     nameIsGivenName },
+    { "family",    nameIsFamilyName },
+    { "universal", nameIsGivenName | nameIsFamilyName },
+    { "backer",    nameIsFullName },
+    { "city",      nameIsTownName },
+    { "world",     nameIsWorldName }
+};
+static const str_flag gender_flags[] = {
+    { "male",   nameIsMaleName },
+    { "female", nameIsFemaleName },
+    { "unisex", nameIsUnisexName }
+};
+
+static nameFlags usage_flag( std::string const &usage )
+{
+    for( auto const &i : usage_flags ) {
+        if( usage == i.str ) {
```

Comment:
`std::find( usage_flags.begin(), usage_flags.end(), usage )`
Though you'd need to change the C-style array to C++ `std::array` or `vector`, `map` etc.